types/src/transaction/mod.rs

```diff
@@ -2607,12 +2607,32 @@ pub trait BlockExecutableTransaction: Sync + Send + Clone + 'static {
 }
 
 pub struct ViewFunctionOutput {
-    pub values: Result<Vec<Vec<u8>>>,
+    pub values: Result<Vec<Vec<u8>>, ExecutionStatus>,
```

Comment:
If the goal is to propagate messages, I think just extracting the string is easier and makes this independent of all partial errors and ugly statuses which discard many messages and remap error codes to weird things :(